redash/handlers/data_sources.py

```diff
@@ -34,12 +34,15 @@ def get(self):
 
 
 class DataSourceResource(BaseResource):
-    @require_admin
     def get(self, data_source_id):
```

Comment:
It should still require some permission to view the data source details, because otherwise this API will be available when using the query API key...

Also, when the user doesn't have a `list_data_sources` permission it's very likely the admin wouldn't want them to see the details of the data source anyway.

Let's keep this new implementation, but still require `list_data_sources` permission here for now (ideally we would use a new permission, but this is a bit of a hassle to add).

Reply:
I suggested the same but late a bit :sweat_smile:

Reply:
Having "list_data_sources" doesn't fix the problem :(

> Also, when the user doesn't have a list_data_sources permission it's very likely the admin wouldn't want them to see the details of the data source anyway

I had that in mind, that's why I put the "all" only when the user is an admin.

https://github.com/getredash/redash/blob/8fb5bc785a3346920a65a99f2da1e4522d0b2e92/redash/handlers/data_sources.py#L42

Basically we show exactly the same information we would show in the list resource if the user is not an admin.

I see your point anyway, just that if felt ok as the user would actually be "listing data sources". WDYT about only returning the `{ view_only: true/false }` when the user doesn't have the list permission?

Reply:
4958ec404b41532c31b026f7857d2b4ceb676a52 with "returning the `{ view_only: true/false }` when the user doesn't have the list permission"

Reply:
> I had that in mind, that's why I put the "all" only when the user is an admin.

By details I meant even the name or type.